crates/rpc-types/src/error.rs

```diff
@@ -101,3 +101,19 @@ pub enum InvalidInboxEntry {
     #[error("underlying database has I/O issues or is corrupted")]
     DataCorruption = -321501,
 }
+
+#[cfg(feature = "jsonrpsee")]
+impl From<SuperchainDAError> for jsonrpsee::types::ErrorObjectOwned {
+    fn from(err: SuperchainDAError) -> Self {
+        jsonrpsee::types::ErrorObjectOwned::owned(err as i32, err.to_string(), None::<()>)
+    }
+}
+
+#[cfg(feature = "jsonrpsee")]
+impl TryFrom<jsonrpsee::types::ErrorObjectOwned> for SuperchainDAError {
+    type Error = derive_more::TryFromReprError<i32>;
+
+    fn try_from(err: jsonrpsee::types::ErrorObjectOwned) -> Result<Self, Self::Error> {
+        err.code().try_into()
+    }
+}
```

Comment:
do we need this conversion?

Reply:
good catch, no not really since reth is using alloy generic rpc error I see now